src/lambda_handler.py

```diff
@@ -1,89 +1,207 @@
-    if character not in cowsay.characters.CHARS.keys():
-        error_msg = 'Character "%s" is not valid.'
-        logging.error(error_msg, character)
-        failed_task(result, error_msg % character)
     else:
-        contents: str = event.get("contents", "Hello from AWS Lambda!")
-        logger.info(
-            'Creating output using "%s" with contents "%s"', character, contents
+        version = "V3" if "baseMetricV3" in cve["impact"] else "V2"
+        cvss_base_score = cve["impact"]["baseMetric" + version]["cvss" + version][
+            "baseScore"
+        ]
+        cvss_version_temp = cve["impact"]["baseMetric" + version]["cvss" + version][
+            "version"
+        ]
+
+        # Fill document fields with CVE data
+        print(".", end="")
+
+        print("Adding CVE: ", cve_id, " to database")
+
+        cve_doc = CVEDoc(
+            id=cve_id, cvss_score=float(cvss_base_score), cvss_version=cvss_version_temp
         )
-        result["message"] = cowsay.get_output_string(character, contents)
 
-    return result
+        if not cve_id:
+            raise ValueError("JSON does not look like valid CISA CVE data.")
 
+        await cve_doc.save()
+        return cve_doc.id
 
-def task_divide(event) -> dict[str, Union[Optional[float], bool]]:
-    """Divide one number by another and provide the result."""
-    result: dict[str, Union[Optional[float], bool]] = {"result": None, "success": True}
-    numerator: str = event.get("numerator", None)
-    denominator: str = event.get("denominator", None)
 
-    if denominator is None or numerator is None:
-        error_msg = "Request must include both a numerator and a denominator."
-        logging.error(error_msg)
-        failed_task(result, error_msg)
-    else:
-        try:
-            variable_error_msg = "numerator: %s, denominator: %s"
-            result["result"] = example_div(int(numerator), int(denominator))
-        except ValueError:
-            error_msg = "The provided values must be integers."
-            logging.error(error_msg)
-            logging.error(variable_error_msg, numerator, denominator)
-            failed_task(result, error_msg)
-        except ZeroDivisionError:
-            error_msg = "The denominator cannot be zero."
-            logging.error(error_msg)
-            logging.error(variable_error_msg, numerator, denominator)
-            failed_task(result, error_msg)
-
-    return result
-
-
-def handler(event, context) -> dict[str, Optional[str]]:
+async def process_cve_json(json_stream) -> None:
+    """Process the provided CVEs JSONs and update the database with its contents."""
+    data = json.load(json_stream)
+
+    if data.get("CVE_data_type") != "CVE":
+        raise ValueError("JSON does not look like valid NVD CVE data.")
+
+    tasks = [asyncio.create_task(process_nvd(cve)) for cve in data.get("CVE_Items", [])]
+
+    for task in asyncio.as_completed(tasks):
+        await task
```

Comment:
Would it be simpler to use a `TaskGroup` here?

Reply:
Task groups were I believe introduced in [Python 3.11](https://www.python.org/downloads/release/python-3110/) no? Until the way we do Lambdas is updated (work in progress) the repos are still designed to support multiple Python versions.

Reply:
This has been fixed in the latest push.